packages/grid/_modules_/grid/hooks/features/columns/useColumns.ts

```diff
@@ -57,7 +61,7 @@ function hydrateColumns(
   logger: Logger,
 ): Columns {
   logger.debug('Hydrating Columns with default definitions');
-  let mappedCols = mapColumns(columns, columnTypes, viewportWidth);
+  let mappedCols = mapColumns(columns, columnTypes, viewportWidth, withCheckboxSelection);
   if (withCheckboxSelection) {
     mappedCols = [checkboxSelectionColDef, ...mappedCols];
```

Comment:
Why don't you swap this row with row 60 so when you do check the columns you already have the checkbox added?

Reply:
I removed the `mapColumns` function, it's not needed anymore. Now the logic is contained within the `hydrateColumns `.